src/sage/schemes/curves/plane_curve_arrangement.py

```diff
@@ -0,0 +1,1246 @@
+                dic1[k] = dic[k]
+            dic1[j] = dic[n - 1]
+            for k in range(j + 1, n):
+                dic1[k] = dic[k - 1]
+        elif infinity_divides:
+            dic1 = {k: dic[k] for k in range(n)}
+            dic1[j] += dic[n]
+        else:
+            dic1 = dic
+        if not simplified:
+            self._fundamental_group = G
+            self._meridians = dic1
+        else:
+            self._fundamental_group_simplified = G
+            self._meridians_simplified = dic1
+        return G
+
+    def meridians(self, simplified=True):
+        r"""
+        Meridians of each irreducible component.
+
+        OUTPUT:
+
+        A dictionary which associates the index of each curve with
+        its meridians, including the line at infinity if it can be
+        computed
+
+        EXAMPLES::
+
+            sage: # needs sirocco
+            sage: H.<x, y, z> = ProjectivePlaneCurveArrangements(QQ)
+            sage: A = H(y^2 + x*z, y + x - z, x)
+            sage: A.fundamental_group().sorted_presentation()
+            Finitely presented group < x0, x1 | x1^-1*x0^-1*x1*x0 >
+            sage: A.meridians()
+            {0: [x1], 1: [x0], 2: [x1^-1*x0^-1*x1^-1]}
+            sage: A = H(y^2 + x*z, z, x)
+            sage: A.fundamental_group()
+            Finitely presented group < x0, x1 | (x1*x0)^2*(x1^-1*x0^-1)^2 >
+            sage: A.meridians()
+            {0: [x0, x1*x0*x1^-1], 1: [x0^-1*x1^-1*x0^-1], 2: [x1]}
+            sage: A = H(y^2 + x*z, z*x, y)
+            sage: A.fundamental_group()
+            Finitely presented group < x0, x1, x2 | x2*x0*x1*x0^-1*x2^-1*x1^-1,
+                                                    x1*(x2*x0)^2*x2^-1*x1^-1*x0^-1*x2^-1*x0^-1 >
+            sage: A.meridians()
+            {0: [x0, x2*x0*x2^-1], 1: [x2, x0^-1*x2^-1*x1^-1*x0^-1], 2: [x1]}
+
+        This function needs
+        :func:`ProjectivePlaneCurveArrangements.fundamental_group`
+        with the same options, where some examples are shown.
+
+        .. WARNING::
+
+            This functionality requires the sirocco package to be installed.
+        """
+        if not simplified:
+            computed = bool(self._meridians)
+        else:
+            computed = bool(self._meridians_simplified)
+        if not computed:
+            _ = self._fundamental_group(simplified=simplified)
+        if not simplified:
+            return self._meridians
+        return self._meridians_simplified
+
+
+class AffinePlaneCurveArrangements(Parent, UniqueRepresentation):
+    """
+    Affine curve arrangements.
+
+    INPUT:
+
+    - ``base_ring`` -- ring; the base ring
+
+    - ``names`` -- tuple of strings; the variable names
+
+    EXAMPLES::
+
+        sage: H.<x, y> = AffinePlaneCurveArrangements(QQ)
+        sage: H(x, y^2, x-1, y-1)
+        Arrangement (x, y^2, x - 1, y - 1) in Affine Space
+        of dimension 2 over Rational Field
+    """
+    Element = AffinePlaneCurveArrangementsElement
+
```

Comment:
I do not know exactly what to do at this moment, I will take care of it later

Reply:
You can use the function
```
from sage.structure.category_object import normalize_names
```
and preprocessing it with `__classcall__` will mean it works with `'x,y,z'` and `['x', 'y', 'z']` and `('x', 'y', 'z')` and returns the same object.

Reply:
I tried something but I am not sure if it is what it should be done.

Reply:
I added the function `__classcall__`, see line 963, importing `normalize_names`. It works in this class, I tried to do the same for `ProjectivePlaneCurveArrangements` but I kept the old code since it did not work

Reply:
Note that `__classcall__` is also called by subclasses, so the arguments need to be consistent (or at least handled/passed appropriately). You can use `__classcall_private__` if you want it to apply to just that single class (and not subclasses).

Reply:
I found the mistake